 design/detailed-diagrams.md

```diff
@@ -0,0 +1,686 @@
+  NetC -->|Dependency| NetB
+end
+
+subgraph Js["Javascript"]
+  JsA["A<sub>js</sub>"]
+  JsB["B<sub>js</sub>"]
+  JsC["C<sub>js</sub>"]
+
+  JsA -->|Dependency| JsB
+  JsC -->|Dependency| JsB
+end
+
+Dafny ---> Java
+Dafny ---> Net
+Dafny ---> Js
+```
+
+
+## Smithy Dafny
+
+### Library Development
+
+Here is a example of the various components
+of smithy dafny fit together to translate a library.
+
+
+```mermaid
+---
+title: How smithy-dafny packages a Dafny Implementation
+---
+graph LR
+
+subgraph Smithy
+  AModel["Model for A"]
+end
+
+subgraph Dafny
+  subgraph A["Package for A"]
+    ATypes["Dafny Types for A"]
+    AInterface["Dafny Interface for A"]
+    AImplementation["Dafny implementation of A"]
+
+    AImplementation --> AInterface
+    ATypes --> AInterface
+    ATypes --> AImplementation
+
+    AModel -.->|CodeGen| ATypes
+    AModel -.->|CodeGen| AInterface
+  end
+
+  subgraph ATests["Tests for A"]
+    AUnitTests["Unit Tests for A"]
+    AIntegrationTests["Integration Tests for A"]
+
+    AUnitTests --> |Testing| AImplementation
+    AIntegrationTests -->|Testing| AInterface
+  end
+end
+
+subgraph Net[".Net"]
+  subgraph NetA["A<sub>.net</sub>"]
+    NetAFromDafny
+    NetATypes
+    NetAInterface
+    NetATypeConversion["To/From Dafny"]
+
+    NetATypeConversion --> NetAInterface
+    NetAFromDafny --> NetAInterface
+    NetATypes --> NetAInterface
+    NetATypes --> NetATypeConversion
+    AModel -.->|CodeGen| NetATypes
+    AModel -.->|CodeGen| NetAInterface
+    AModel -.->|CodeGen| NetATypeConversion
+    A ==>|Dafny Compilation| NetAFromDafny
+  end
+
+  NetATests
+
+  ATests ==>|Dafny Compilation| NetATests
+  NetATests -->|Testing| NetAFromDafny
+end
+```
+
+### Dependencies
+
+This is a description of how Smithy Dafny
+handles dependencies.
+
+```mermaid
+---
+title: How Smithy Dafny handles dependencies
+---
+graph LR
+
+subgraph Smithy
+  AModel["Model for A"]
+  BModel["Model for B"]
+
+  BModel --> AModel
+end
+
+subgraph Dafny
+  subgraph A["Package for A"]
+    ATypes["Dafny Types for A"]
+    AInterface["Dafny Interface for A"]
+    AImplementation["Dafny implementation of A"]
+
+    AImplementation --> AInterface
+    ATypes --> AInterface
+    ATypes --> AImplementation
+
+    AModel -.->|CodeGen| ATypes
+    AModel -.->|CodeGen| AInterface
+    BModel -.->|CodeGen| ATypes
+  end
+
+  B["Package for B"]
+  B --> ATypes
+  BModel -.->|CodeGen| B
+
+end
+
+subgraph Net[".Net"]
+  subgraph NetA["A<sub>.net</sub>"]
+    NetAFromDafny
+    NetATypes
+    NetAInterface
+    NetATypeConversion["To/From Dafny"]
+
+    NetATypeConversion --> NetAInterface
+    NetAFromDafny --> NetAInterface
+    NetATypes --> NetAInterface
+    AModel -.->|CodeGen| NetATypes
+    AModel -.->|CodeGen| NetAInterface
+    AModel -.->|CodeGen| NetATypeConversion
+    A ==>|Dafny Compilation| NetAFromDafny
+
+    ATypes ~~~ NetAInterface
+  end
+
+  NetB["B<sub>.net</sub>"]
+  B ==>|Dafny Compilation| NetB
+
+  NetB --> NetAFromDafny
```

Comment:
it's not clear, but this is saying that the "internals" of the .NET B project call into "internals" of the .NET A project?